### PR TITLE
Improvement: Added open-needs service

### PR DESCRIPTION
**Pull request by iSOLveIT**
The open-needs service will allow users to read/import `need` objects from an Open-Needs Server.

### CODE REVIEW

File: docs/roles.rst [hunk 1]
```diff
@@ -31,6 +31,8 @@ Example
    :value: 20
    :unit: slices
 
+.. _needref:
```
Comment (danwos) [inline docs/roles.rst:34]: Does this make sense here?

If yes, then a second ``:`` is missing at the end.

**Review by danwos — CHANGES_REQUESTED** [review]
Really great work and good test cases (incl. the monkeypatch part).
Only a small finding in the docs.

Overall next time pls try to separate docs-enhancements and feature development :)
It's always hard to make reviews of huge PRs.

File: docs/roles.rst (hunk 1 above)
Comment (iSOLveIT) [reply docs/roles.rst:34]: This

> Does this make sense here?
> 
> If yes, then a second `:` is missing at the end.

Yes, it makes sense. It creates a link to that section so a second `:` is not needed.

**Comment by iSOLveIT** [comment]
> Really great work and good test cases (incl. the monkeypatch part). Only a small finding in the docs.
> 
> Overall next time pls try to separate docs-enhancements and feature development :) It's always hard to make reviews of huge PRs.

The documentation hasn't been added yet. The current content is just to test out the output for the feature.
I am now writing the complete docs for the feature.

File: docs/roles.rst (hunk 1 above)
Comment (danwos) [reply docs/roles.rst:34]: Argh, sure. My fault.
Was thinking on a directive and not the basic sphinx reference feature.
Sorry :)

**Review by danwos — APPROVED** [review]

**Pull request merged**